pkg/controller/directvolumemigration/stunnel.go

```diff
@@ -0,0 +1,495 @@
+					"owner":   "directvolumemigration",
+				},
+				Type: corev1.ServiceTypeClusterIP,
+			},
+		}
+		volumes := []corev1.Volume{
+			{
+				Name: "stunnel-conf",
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: "directvolumemigration-stunnel-config",
+						},
+					},
+				},
+			},
+			{
+				Name: "stunnel-certs",
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: "directvolumemigration-stunnel-certs",
+						Items: []corev1.KeyToPath{
+							{
+								Key:  "tls.crt",
+								Path: "tls.crt",
+							},
+							{
+								Key:  "ca.crt",
+								Path: "ca.crt",
+							},
+							{
+								Key:  "tls.key",
+								Path: "tls.key",
+							},
+						},
+					},
+				},
+			},
+		}
+		trueBool := true
+		runAsUser := int64(0)
+		containers := []corev1.Container{}
+
+		containers = append(containers, corev1.Container{
+			Name:    "stunnel",
+			Image:   "quay.io/konveyor/rsync-transfer:latest",
+			Command: []string{"/bin/stunnel", "/etc/stunnel/stunnel.conf"},
+			Ports: []corev1.ContainerPort{
+				{
+					Name:          "stunnel",
+					Protocol:      corev1.ProtocolTCP,
+					ContainerPort: int32(2222),
+				},
+			},
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      "stunnel-conf",
+					MountPath: "/etc/stunnel/stunnel.conf",
+					SubPath:   "stunnel.conf",
+				},
+				{
+					Name:      "stunnel-certs",
+					MountPath: "/etc/stunnel/certs",
+				},
+			},
+			SecurityContext: &corev1.SecurityContext{
+				Privileged:             &trueBool,
+				RunAsUser:              &runAsUser,
+				ReadOnlyRootFilesystem: &trueBool,
+			},
+		})
+
+		labels := map[string]string{
+			"app":     "directvolumemigration-rsync-transfer",
+			"owner":   "directvolumemigration",
+			"purpose": "stunnel",
+		}
+		clientPod := corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "directvolumemigration-stunnel-transfer",
+				Namespace: ns,
+				Labels:    labels,
+			},
+			Spec: corev1.PodSpec{
+				Volumes:    volumes,
+				Containers: containers,
+			},
+		}
+		err := srcClient.Create(context.TODO(), &svc)
+		if k8serror.IsAlreadyExists(err) {
+			t.Log.Info("Stunnel client svc already exists on source", "namespace", svc.Namespace)
+		} else if err != nil {
+			return err
+		}
+		t.Log.Info("stunnel client svc created", "name", clientPod.Name, "namespace", svc.Namespace)
+		err = srcClient.Create(context.TODO(), &clientPod)
+		if k8serror.IsAlreadyExists(err) {
+			t.Log.Info("Stunnel client pod already exists on source", "namespace", clientPod.Namespace)
+		} else if err != nil {
+			return err
+		}
+		t.Log.Info("stunnel client pod created", "name", clientPod.Name, "namespace", clientPod.Namespace)
+	}
+	return nil
+}
+
+// check if stunnel client pods are running
+func (t *Task) areStunnelClientPodsRunning() (bool, error) {
+	// Get client for destination
```

Comment:
after a certain phase, we probably want this function to be executed on every reconcile loop